test-e2e/roles/test_syncthing_cluster_sync/tasks/main.yml

```diff
@@ -83,47 +121,26 @@
   vars:
     data: 'hello from syncthing-1'
     path: /testdata.txt
-    pvc_name: "{{ syncthings[0] }}"
+    pvc_name: "{{ syncthings[0].Name }}"
+    pvc_namespace: "{{ syncthings[0].Namespace }}"
 
 - name: Ensure data was synchronized to other instances
   include_role:
     name: pvc_has_data
   vars:
     data: 'hello from syncthing-1'
     path: /testdata.txt
-    pvc_name: "{{ item }}"
+    pvc_name: "{{ item.Name }}"
+    pvc_namespace: "{{ item.Namespace }}"
     timeout: 300
   loop: "{{ syncthings[1:] }}"
 
-- name: Find syncthing pods
-  kubernetes.core.k8s_info:
-    api_version: v1
-    kind: Pod
-    namespace: "{{ namespace }}"
-    label_selectors:
-      - "app.kubernetes.io/component=syncthing-mover"
-  register: st_pods
-
 - name: Kill all syncthing pod instances
-  kubernetes.core.k8s:
-    state: absent
-    api_version: v1
-    kind: Pod
-    name: "{{ item.metadata.name }}"
-    namespace: "{{ namespace }}"
-  loop: "{{ st_pods.resources }}"
-
-- name: Wait for pods to be gone
-  kubernetes.core.k8s:
-    state: absent
-    api_version: v1
-    kind: Pod
-    name: "{{ item.metadata.name }}"
-    namespace: "{{ namespace }}"
-    wait: true
-    wait_timeout: 600
-  timeout: 600
-  loop: "{{ st_pods.resources }}"
+  include_tasks:
+    file: kill_pods_by_label.yml
+  loop: "{{ syncthings | map(attribute='Namespace') | list }}"
+  loop_control:
+    loop_var: st_namespace
```

Comment:
I actually didn't realize it was parallel previously!  I haven't seen this part take all that long (although it may take longer in those CI envs), since there are only 3 in this test.

I think there is a potential issue (i've only seen this 1x in my local env) where waiting for the pod to be deleted may miss it, since it gets started before we catch the pod being gone.  Think this is a problem whether we do serial or parallel though.